### ISSUE TITLE
enhancement, ego planner + curobo

### ISSUE THREAD

**Issue by WarriorHanamy** [WarriorHanamy/realworld_modules#38]

**Comment by WarriorHanamy** [comment]
1. depth + depth_raw (cpu)
2,. qu jibian + timestamp synced + imu
3. downstreaming, and filtering (statictics)
4. pointclouds, depth, sparse point.  VIO to obtain velocity.
5. local-env modeling.
6. cpu path finding
7. cpu trajectory optimization.
8. replan + fsm.

**Comment by WarriorHanamy** [comment]
1. depth + depth_raw (cpu)
2,. qu jibian + timestamp synced + imu
3. downstreaming, and filtering (statictics)
4. pointclouds, depth, sparse point.  VIO to obtain velocity.
5. local-env modeling.
6. cpu path finding
7. cpu trajectory optimization.
8. replan + fsm.

**Comment by WarriorHanamy** [comment]
External solvers (curobo._src.optim.external) – delegate the outer loop to [ScipyOpt](https://nvlabs.github.io/curobo/latest/api/curobo.optim.html#curobo.optim.ScipyOpt) (SciPy) or [TorchOpt](https://nvlabs.github.io/curobo/latest/api/curobo.optim.html#curobo.optim.TorchOpt) (torch.optim) while still evaluating costs and gradients on the GPU through the cuRobo rollout.
